### DISCUSSION TITLE
Ability to "Ungroup" tabs

### DISCUSSION THREAD

**Discussion by jsmorabito** [oxdc/obsidian-vertical-tabs#67] [Ideas]
I would like to be able to ungroup tabs that are grouped together. One option to do this could be by dragging the tab out of the group. Another option could be in the context menu when you right click on a group.

Thanks for the great plugin. Really enjoying it!

**Comment by oxdc** [comment]
Hi @jsmorabito ! I've converted your FR to this discussion. Sorry for my late reply. I've been busy these days. As you may have noticed, *Vertical Tabs* groups tabs by leveraging Obsidian's native support for workspace splits. This design is to ensure that the states are synced with Obsidian's UI. So, the major issue of "ungrouping" a group is: where do these tabs go exactly? There might not be a preferred location for them. May I ask: if you can multi-select and drag the tabs to another group, making the current group disappear, is this solution satisfactory for you?

**Reply by jsmorabito** [reply]
Thanks for opening the discussion 😁 Ah good point, I overlooked fact of workspace splits being leveraged...maybe there is a creative solution...I will think on this. I do believe there is a valid use case for the state of having ungrouped tabs. I'll circle back once I've given it some thought.

Your suggested workaround does support the scenario where one might want to have no groups present, but the use case I am looking to support is having groups and ungrouped tabs simultaneously. There are occasions where I want to open a file and due to the current nature of the plugin, it can only be added to a group when I am using them, and this creates a bit of friction when my intention is not to categorize the note I'm opening.

**Reply by jsmorabito** [reply]
![image](https://github.com/user-attachments/assets/77d6e98d-65a9-4805-b3f2-303c74d55dfe)

This is how I'm imagining it from the user side, but I'm struggling to come up with an elegant way to handle the constraint of being tied to workspace splits. In the screenshot "Tab group" is what is presently called "Grouped tabs" and represents all tabs in a split.

I think the only way it could work is if 'zen mode' and 'show active tabs' were always active. That way a user who has multiple, single "ungrouped" tabs wouldn't see multiple workspace splits at the same time which is how I imagine that would be handled. Let me know if that makes sense or if I should elaborate on any of it.

**Reply by oxdc** [reply]
> having groups and ungrouped tabs simultaneously

Hi @jsmorabito ! Thank you! Logically, "ungrouped tabs" are tabs in a "default group". So, I think this use case is not excluded from the current design.

> There are occasions where I want to open a file and due to the current nature of the plugin, it can only be added to a group when I am using them, and this creates a bit of friction when my intention is not to categorize the note I'm opening.

You don't have to categorize them. Just pick a random group, maybe the first group, to be your default group. You don't have to rename it to "Default group"! Just let it be there! In fact, *Vertical Tabs* **discourages** you to spend time deciding and categorizing tabs, which is a waste of time and mind power.

The correct way to use groups is like this: When you have made some progress on Project A and you need to move to another, say Project B. Just click the "New group" button to create a new group for tabs of Project B. If you will be confused in the future, rename the groups to "Project A" and "Project B". If not, renaming is optional.

Another correct use: You can put all tabs in some default group without thinking about their categorizes or scopes.

The incorrect use is like this: Every time you open a note, you spend a fraction of time determining which group it fits best. You ask yourself questions like:
- "Should I put it in a new group?"
- "Maybe both this group and that group are appropriate? Should I duplicate this tab?"
- "Should I organize groups based on responsibility or projects or something else?"

Things get worse if we have grouped and ungrouped tabs and tree-like structures:
- "Should I put this tab in some group or leave it as ungrouped?"
- "The tree structure is starting getting complicated. Maybe I need to spend some time reorganize it?"

**Reply by jsmorabito** [reply]
Thank you for the response and explanation 🙇. I do appreciate your taking the time to describe the preferred workflows of the plugin, but I don't think they are compatible with my ideal use case and that's totally ok. I understand the constraints for why it behaves how it does currently and it's still a very useful plugin for me!

Feel free to resolve or close this discussion if you like or leave it open in case others might want to share their own use cases.

Thanks again for making this!

**Reply by oxdc** [reply]
Hi @jsmorabito ! Sorry for my late reply. I've been trying hard to keep everything on track. I will gradually roll out some new features in the next few months. You may see if these improvements can be adapt to your workflow. Besides, please feel free to share your thoughts, especially on why you want this feature. I'll leave this discussion open.

**Comment by oxdc** [comment]
See #126. This feature is now available to [Beta Program](https://vertical-tabs-docs.oxdc.dev/Beta-Versions/beta-program) subscribers and will be released publicly later.

New command: "Vertical Tabs: Close current editor group"
